src/gt4py/next/utils.py

```diff
@@ -388,3 +390,8 @@ def canonicalize_call_args(
     """
 
     return make_args_canonicalizer_for_function(func)(*args, **kwargs)
+
+
+class SequentialPrefixedIDGenerator(eve_utils.CustomDefaultDictBase):
```

Comment:
```suggestion
class SequentialPrefixedIDGenerator(eve_utils.CustomDefaultDictBase[str, eve_utils.SequentialIDGenerator]):
```

Maybe something like:
`SequentialIDGeneratorPool`
`CollectiveSequentialIDGenerator`
`SharedSequentialIDGenerators`
The sequential term is also not essential in my opinion as we only care about uniqueness, and since this is in gt4py utils not eve, so close to the caller not the implementation of the generator I would personally go for `IDGeneratorPool`. I'm also fine with the existing name, but it took me a while to understand that there is `SequentialIDGenerator` and `SequentialPrefixedIDGenerator` and which one is which. If you like the original name most, how about appending an `s? That would be easy to memorize.